taf/git.py

```diff
@@ -150,15 +150,32 @@ def remotes(self) -> List[str]:
         return self._remotes
 
     @property
-    def is_git_repository_root(self) -> bool:
-        """Check if path is git repository."""
-        git_path = self.path / ".git"
-        return self.is_git_repository and (git_path.is_dir() or git_path.is_file())
+    def is_git_repository(self) -> bool:
+        try:
+            repo = pygit2.Repository(str(self.path))
+            if self.is_bare_repository and not repo.is_bare:
+                return False
+            if not self.is_bare_repository and repo.head is None:
+                return False
+            return True
+        except (KeyError, ValueError, OSError, pygit2.GitError):
+            return False
 
     @property
-    def is_git_repository(self) -> bool:
-        repo_path = pygit2.discover_repository(str(self.path))
-        return repo_path is not None
+    def is_git_repository_root(self) -> bool:
+        if not self.is_git_repository:
+            return False
+        try:
+            repo = pygit2.Repository(str(self.path))
+            if self.is_bare_repository:
+                return repo.is_bare and Path(repo.path).resolve() == self.path.resolve()
```

Comment:
I really am not sure that this checks are exactly what we want here. It would be easy to tests if we actually had a test. There is a `test_git.py` test which we just recently added and only contains one tests. Could you add a new test, where you'd initialize a bare repository (using this module's `init` functionality`) then validate these changes? We should also test the case when the repository is not bare